### ISSUE TITLE
CLI: Add support for sentry

### ISSUE THREAD

**Issue by rajdip-b** [keyshade-xyz/keyshade#529]
## Description

We would like the CLI to send errors to Sentry if `metrics_enabled` (refer to #528) is enabled. Sentry should be notified in the following events:
- An internal error occurs in any of the components of the CLI
- A 500 is returned from the API

## Solution

- You would need to send the error to sentry with the environment set as `CLI`.
- Add a `Logger.report` function that will be used to first call `Logger.error` and then send the data to sentry. We should not be printing too much using `Logger.error` in this case.

**Requires #528**

**Comment by rajdip-b** [comment]
Sure!, but it depends on the #528 so we need that before you an attempt this.

**Comment by rajdip-b** [comment]
@muntaxir4 can you get started with this?

**Comment by muntaxir4** [comment]
Yes picked this up today. Should the DSN be restricted to Keyshade only?

**Comment by rajdip-b** [comment]
Yeah. Dont worry too much about that though. I'm looking into these changes. Perhaps don't raise a pr aswell. I'm also working on a lot of changes to the release system. It would be better if your put up a pr to that other branch. I'll specify you the name once I'm done with that.

**Comment by muntaxir4** [comment]
> Yeah. Dont worry too much about that though. I'm looking into these changes. Perhaps don't raise a pr aswell. I'm also working on a lot of changes to the release system. It would be better if your put up a pr to that other branch. I'll specify you the name once I'm done with that.

Can you specify which branch it is?

**Comment by rajdip-b** [comment]
Couldn't yet put that up. You can continue your work on some other branch. And when I'm done, you can just raise a PR into that branch.
